eventhubs/data-plane/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/EventDataImpl.java

```diff
@@ -270,4 +271,17 @@ public int compareTo(EventData other) {
                 other.getSystemProperties().getSequenceNumber()
         );
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        EventDataImpl eventData = (EventDataImpl) o;
+        return Objects.equals(systemProperties, eventData.systemProperties);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(systemProperties);
+    }
```

Comment:
replaced systemProperties with bodyData, in the EventData class it mentioned "While using client libraries released by Microsoft Azure Eventhubs, the getBytes() which corresponds to Binary variable bodyData and getObject() which corresponds to Object amqpBody",  alone are sufficient. So I picked bodyData, since bodyData has higher priority than amqpBody

Reply:
Why did you not use both `bodyData` and `amqpBody`? Even properties might be useful? I don't understand where the statement about "bodyData has higher priority than amqpBody" is coming from?

Reply:
if (this.bodyData != null) {
            amqpMessage.setBody(new Data(this.bodyData));
        } else if (this.amqpBody != null) {
            if (this.amqpBody instanceof List) {
                amqpMessage.setBody(new AmqpSequence((List) this.amqpBody));
            } else {
                amqpMessage.setBody(new AmqpValue(this.amqpBody));
            }
        }

Reply:
You should have both in here, not just bodyData.